pkg/visor/hypervisor.go

```diff
@@ -710,7 +716,18 @@ func (hv *Hypervisor) putApp() http.HandlerFunc {
 			}
 		}
 
-		httputil.WriteJSON(w, r, http.StatusOK, ctx.App)
+		if err := ctx.API.SetAppDetailedStatus(ctx.App.Name, vpn.ClientStatusConnecting); err != nil {
```

Comment:
There is a small delay when we send back the response of the API `putApp` and the app starting and changing the the `detailed_status` by itself. That used to cause the vpn-ui to show disconnecting first and then connecting after we started it.
Another way would be to send a `starting` detailed status regardless of the apps state and it will update to the correct status after a while. But that makes the other apps appear slow so I went with this approach.